packages/react-storage/src/components/StorageBrowser/Views/Controls/Table.tsx

```diff
@@ -1,7 +1,135 @@
+
 import { StorageBrowserElements } from '../../context/elements';
 
-export interface TableControl<T extends StorageBrowserElements>
-  extends Pick<
+const {
+  Table: BaseTable,
+  TableBody: BaseTableBody,
+  TableData: BaseTableData,
+  TableHead: BaseTableHead,
+  TableHeader: BaseTableHeader,
+  TableRow: BaseTableRow,
+  Button,
+  Icon,
+} = StorageBrowserElements;
+
+const BLOCK_NAME = 'table';
+
+const Table = withBaseElementProps(BaseTable, {
+  className: `${BLOCK_NAME}`,
+});
+
+const TableBody = withBaseElementProps(BaseTableBody, {
+  className: `${BLOCK_NAME}__body`,
+});
+
+const TableHead = withBaseElementProps(BaseTableHead, {
+  className: `${BLOCK_NAME}__head`,
+});
+
+const TableHeader = withBaseElementProps(BaseTableHeader, {
+  className: `${BLOCK_NAME}__header`,
+});
+
+const TableHeaderButton = withBaseElementProps(Button, {
+  className: `${BLOCK_NAME}__header-button`,
+});
+
+const TableData = withBaseElementProps(BaseTableData, {
+  className: `${BLOCK_NAME}__data`,
+});
+
+const TableDataButton = withBaseElementProps(Button, {
+  className: `${BLOCK_NAME}__data-button`,
+});
+
+const TableRow = withBaseElementProps(BaseTableRow, {
+  className: `${BLOCK_NAME}__row`,
+});
+
+const iconAttributes = {
+  'aria-hidden': true,
+  width: '24',
+  height: '24',
+  viewBox: '0 0 24 24',
+  fill: 'none',
+  xmlns: 'http://www.w3.org/2000/svg',
+};
+
+const SortIndeterminateIcon = withBaseElementProps(Icon, {
+  children: (
+    <path
+      fillRule="evenodd"
+      clipRule="evenodd"
+      d="M4.0001 4C3.20044 4 2.72425 4.89205 3.16923 5.55646L7.16674 11.5253C7.56328 12.1174 8.43424 12.1165 8.8296 11.5236L12.8098 5.55479C13.2529 4.89025 12.7765 4 11.9778 4H4.0001ZM10.1057 6H5.87278L7.99999 9.17845L10.1057 6Z"
+      fill="#16191F"
+    />
+  ),
+  className: `${BLOCK_NAME}__sort-icon--indeterminate`,
+  ...iconAttributes,
+});
+
+const SortAscendingIcon = withBaseElementProps(Icon, {
+  children: (
+    <path
+      fillRule="evenodd"
+      clipRule="evenodd"
+      d="M3.16512 10.4442C2.72085 11.1086 3.19712 12 3.99642 12H12.0037C12.8021 12 13.2785 11.1104 12.8361 10.4458L8.84116 4.44491C8.44589 3.85117 7.5739 3.85029 7.17744 4.44324L3.16512 10.4442Z"
+      fill="#16191F"
+    />
+  ),
+  className: `${BLOCK_NAME}__sort-icon--ascending`,
+  ...iconAttributes,
+});
+
+const SortDescendingIcon = withBaseElementProps(Icon, {
+  children: (
+    <path
+      transform="rotate(180 12 12)"
+      fillRule="evenodd"
+      clipRule="evenodd"
+      d="M3.16512 10.4442C2.72085 11.1086 3.19712 12 3.99642 12H12.0037C12.8021 12 13.2785 11.1104 12.8361 10.4458L8.84116 4.44491C8.44589 3.85117 7.5739 3.85029 7.17744 4.44324L3.16512 10.4442Z"
+      fill="#16191F"
+    />
+  ),
+  className: `${BLOCK_NAME}__sort-icon--descending`,
+  ...iconAttributes,
+});
+
+export interface ILocation {
+  /**
+   * Name of the specified location or object.
+   */
+  name: string;
+}
```

Comment:
I wanted to create a base interface for the table where users implementing the TableControl component could extend from it. I wanted to have `name` since this needs to be a button for navigating into the location, does that sound right?